template.yaml

```diff
@@ -0,0 +1,984 @@
+      CidrIp:
+        Fn::ImportValue: !Sub ${VpcStackName}-VpcCidr
+      Description: Allow traffic from the VPC on port 80
+      IpProtocol: tcp
+      FromPort: 80
+      ToPort: 80
+
+  DocAppApplicationLoadBalancerSecurityGroupEgress:
+    Type: AWS::EC2::SecurityGroupEgress
+    Properties:
+      GroupId: !GetAtt DocAppApplicationLoadBalancerSecurityGroup.GroupId
+      DestinationSecurityGroupId: !GetAtt DocAppContainerServiceSecurityGroup.GroupId
+      Description: Allow traffic to Container Service
+      IpProtocol: tcp
+      FromPort: 8080
+      ToPort: 8080
+
+  #
+  # API Gateway
+  #
+
+  ApiGateway:
+    Type: AWS::ApiGatewayV2::Api
+    Properties:
+      Name: !Sub "${AWS::StackName}-ApiGateway"
+      ProtocolType: HTTP
+      Tags:
+        Service: Relying Party stub
+        Name: !Sub "${AWS::StackName}-ApiGateway"
+        Source: alphagov/di-auth-stub-relying-party
+        Owner: di-orchestration@digital.cabinet-office.gov.uk
+
+  ApiGatewayIntegration:
+    Type: AWS::ApiGatewayV2::Integration
+    Properties:
+      ApiId: !Ref ApiGateway
+      IntegrationType: HTTP_PROXY
+      ConnectionId:
+        Fn::ImportValue: !Sub "${VpcStackName}-VpcLinkId"
+      ConnectionType: VPC_LINK
+      IntegrationMethod: ANY
+      IntegrationUri: !Ref ApplicationLoadBalancerListener
+      PayloadFormatVersion: 1.0
+
+  ApiRoute:
+    Type: AWS::ApiGatewayV2::Route
+    Properties:
+      ApiId: !Ref ApiGateway
+      RouteKey: "ANY /{proxy+}"
+      Target: !Sub "integrations/${ApiGatewayIntegration}"
+      AuthorizationType: CUSTOM
+      AuthorizerId: !Ref ApiGatewayAuthorizer
+
+  ApiDefaultStage:
+    Type: AWS::ApiGatewayV2::Stage
+    Properties:
+      ApiId: !Ref ApiGateway
+      StageName: "$default"
+      AutoDeploy: true
+      Tags:
+        Service: Relying Party stub
+        Name: !Sub "${AWS::StackName}-ApiDefaultStage"
+        Source: alphagov/di-auth-stub-relying-party
+        Owner: di-orchestration@digital.cabinet-office.gov.uk
+      AccessLogSettings:
+        DestinationArn: !GetAtt ApiGatewayAccessLogsGroup.Arn
+        Format:
+          Fn::ToJsonString:
+            requestId: $context.requestId
+            ip: $context.identity.sourceIp
+            requestTime: $context.requestTime
+            httpMethod: $context.httpMethod
+            path: $context.path"
+            routeKey: $context.routeKey
+            status: $context.status
+            protocol: $context.protocol
+            responseLength: $context.responseLength
+
+  ApiGatewayAccessLogsGroup:
+    Type: AWS::Logs::LogGroup
+    # checkov:skip=CKV_AWS_158:Ensure that CloudWatch Log Group is encrypted by KMS
+    Metadata:
+      checkov:
+        skip:
+          - id: "CKV_AWS_158"
+            comment: "Ensure that CloudWatch Log Group is encrypted by KMS"
+    Properties:
+      LogGroupName: !Sub /aws/apigateway/${AWS::StackName}-ApiGatewayAccessLogsGroup
+      RetentionInDays: 90
```

Comment:
We have 1 day retention for the ECS task. Why the difference?

Reply:
Just copied from various other people and I didn't check they were consistent 😅 happy to go with whichever is best

Reply:
Let's do 90 globally